### ISSUE TITLE
Intel E810 support for sriov-network-metrics-exporter

### ISSUE THREAD

**Issue by dipti262** [k8snetworkplumbingwg/sriov-network-metrics-exporter#21]
Hi,

Could you please let me know if intel e810 NIC is supported in this?
I am currently using the latest commit for sriov-network-metric-exporter (master).

Thanks
Deepti

**Comment by eoghanlawless** [comment]
Hi @dipti262 

The Intel E810 NIC is supported with the latest intel ice drivers.

The readme hadn't been updated when that change was made, it'll be updated soon :)

**Comment by dipti262** [comment]
Hi eoghanlawless,

Thanks for your reply. 
Could you please let me know when the next release will be available (with this intel ice drivers support)?

Thanks
Deepti

**Comment by eoghanlawless** [comment]
We don't have an exact date for the release, should be this month.

The current latest commit on master already supports the latest ice drivers and the E810 NIC.

Thanks,
Eoghan

**Comment by dipti262** [comment]
Hi Eoghan,

Thanks for the response.
We want to test this sriov-network-metrics-exporter on our setup that has Intel E810 NIC.
For this do we need any custom configuration that needs to be done? 

-Deepti

**Comment by dipti262** [comment]
Hi @eoghanlawless 

Above, you have mentioned currently sriov exporter (master supports latest ice driver for intel E810 NIC!
Does it support ice driver 1.9.11 ?

**Comment by dipti262** [comment]
Hi @eoghanlawless 

I am using the sriov-network-exporter (master). 
NIC is E810 driver 
Ice Driver version: 1.9.11 (This is the one which is shipped with SLES 15.4)

We are observing that the vf metrics  appear with 0 value

curl 192.168.165.116:9808/metrics | grep sriov_vf_rx_bytes
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:10:11.0",pf="ens1f2",vf="0"} 0
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:10:11.1",pf="ens1f2",vf="1"} 0
100 27206    0 27206    0     0  4592k      0 --:--:-- --:--:-- --:--:-- 5313k
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:10:11.2",pf="ens1f2",vf="2"} 0
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:10:11.3",pf="ens1f2",vf="3"} 0
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:10:11.4",pf="ens1f2",vf="4"} 0
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:10:11.5",pf="ens1f2",vf="5"} 0
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:10:11.6",pf="ens1f2",vf="6"} 0
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:10:11.7",pf="ens1f2",vf="7"} 0

In the pod logs I see the below messages:
eccd@control-plane-dc286-mast-n3:~> kubectl logs sriov-metric-exporter-pool-worker-4n92x -n monitoring|more
2023/02/02 05:46:46 The kubepoddevice collector is enabled
2023/02/02 05:46:46 The vfstats collector is enabled
2023/02/02 05:46:46 listening on :9808
2023/02/02 05:46:53 using netlink for ens2f0
2023/02/02 05:46:53 PerPF called for ens2f0
2023/02/02 05:46:53 using netlink for ens2f1
2023/02/02 05:46:53 PerPF called for ens2f1
2023/02/02 05:46:53 using netlink for ens10f0
2023/02/02 05:46:53 PerPF called for ens10f0
2023/02/02 05:46:53 using netlink for ens10f2

I checked the interface , I am getting the vf stats in the Netlink collector
```
eccd@pool3-dc286-wk3-srv5-1003:~> ip -s link show ens1f2
5: ens1f2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9000 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether b4:96:91:cb:7c:7a brd ff:ff:ff:ff:ff:ff
    RX:  bytes packets errors dropped  missed   mcast
       1980528   15972      0   15972       0   15972
    TX:  bytes packets errors dropped carrier collsns
             0       0      0       0       0       0
    vf 0     link/ether 02:ff:da:5a:7a:7a brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    RX: bytes  packets  mcast   bcast   dropped
           432        6       0       6        6
    TX: bytes  packets   dropped
             0        0        0
    vf 1     link/ether 02:91:4c:a2:69:42 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    RX: bytes  packets  mcast   bcast   dropped
           432        6       0       6        6
    TX: bytes  packets   dropped
             0        0        0
    vf 2     link/ether 02:cf:7f:4b:74:b9 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    RX: bytes  packets  mcast   bcast   dropped
           432        6       0       6        6
    TX: bytes  packets   dropped
             0        0        0
    vf 3     link/ether 02:48:74:bd:28:99 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    RX: bytes  packets  mcast   bcast   dropped
           432        6       0       6        6
    TX: bytes  packets   dropped
             0        0        0
    vf 4     link/ether 02:38:68:90:a5:bf brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    RX: bytes  packets  mcast   bcast   dropped
           432        6       0       6        6
    TX: bytes  packets   dropped
             0        0        0
    vf 5     link/ether 02:61:a8:d0:20:c7 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    RX: bytes  packets  mcast   bcast   dropped
           432        6       0       6        6
    TX: bytes  packets   dropped
             0        0        0
    vf 6     link/ether 02:72:5d:e5:4f:14 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    RX: bytes  packets  mcast   bcast   dropped
           432        6       0       6        6
    TX: bytes  packets   dropped
             0        0        0
    vf 7     link/ether 02:ce:3e:11:33:dc brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    RX: bytes  packets  mcast   bcast   dropped
           432        6       0       6        6
    TX: bytes  packets   dropped
             0        0        0
    altname enp16s0f2
eccd@pool3-dc286-wk3-srv5-1003:~>
```

Are we missing some configuration that needs to be done? Could you please help us on this.

**Comment by Eoghan1232** [comment]
Hi @dipti262,

that is strange, I did a quick test locally using master and the vf stats were 0 also.
We do have a current release being reviewed, PR #22 .
The vf stats are working correctly in that, I also tested this.

```
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:4b:01.0",pf="ens785f0",vf="0"} 1688
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:4b:01.1",pf="ens785f0",vf="1"} 180
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:4b:01.2",pf="ens785f0",vf="2"} 1748
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:4b:01.3",pf="ens785f0",vf="3"} 180
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:4b:01.4",pf="ens785f0",vf="4"} 180
sriov_vf_rx_bytes{numa_node="0",pciAddr="0000:4b:01.5",pf="ens785f0",vf="5"} 180

```

once that PR is merged into master, I will be creating a new release

also, I tested on ice 1.9.11.

for the time being, you could use that release branch as a test until it's merged?
it is stable and working :)

**Comment by dipti262** [comment]
Hi @Eoghan1232 

Thanks for the response.
We have tried the build from the release branch and the metrics are coming.
Could you please let us know the timeline for new release?

Thanks
Deepti

**Comment by Eoghan1232** [comment]
Hi @dipti262, that release branch is currently in review, once merged, a new release will be created :)
in terms of timeline, this could be end of February.

**Comment by dipti262** [comment]
Hi @Eoghan1232 

We are waiting for the above fix to be available as soon as possible. 
Could you please let us know the exact timeline for the new release?

Thanks
Deepti

**Comment by Eoghan1232** [comment]
Hi @dipti262 , the PR is still in review by the community, so I cannot give an exact date.
as mentioned before, a workaround is to work off the release branch, as this will be merged into main soon.
Once the community has reviewed and approved the changes, we will create a release.

**Comment by Eoghan1232** [comment]
Closing as the PR is now merged.
